### DISCUSSION TITLE
Unstablity of Tensor.data()

### DISCUSSION THREAD

**Discussion by Rick0827** [modular/modular#1732] [Q&A] [answered]
Codes with problem:
<img width="328" alt="image" src="https://github.com/modularml/mojo/assets/137137852/1c185c79-3c16-4442-85ca-05802edeed26">
Env: WSL Ubuntu22.04 || mojo 0.7.0
For the output of the loop, expected "1.0 2.0 3.0 4.0", got
<img width="200" alt="image" src="https://github.com/modularml/mojo/assets/137137852/eff57f86-d1b2-46d9-bc3c-175b7f5b10ac">
which is unstable.
But everything seems to be correct in REPL or add "print(sample[i])" to the loop.

**Comment by mikowals** [comment, marked as answer]
`sample` owns the pointer and is destroyed after its last use.  So `free()` is called on the pointer before you read from it.  You can either keep the tensor from being destroyed by using it after you read the pointer or you can take ownership of the pointer with `sample._steal_ptr()`.

**Reply by Rick0827** [reply]
Thanks a lot!
